pkgs/libresoc/nmigen.nix

```diff
@@ -0,0 +1,49 @@
+{
+  lib,
+  fetchFromGitLab,
+  git,
+  python39Packages,
+  symbiyosys,
+  yices,
+  yosys,
+}:
+with python39Packages;
+  buildPythonPackage rec {
+    pname = "nmigen";
+    version = "unstable-2024-03-31";
+    realVersion = "0.3.dev243+g${lib.substring 0 7 src.rev}";
+
+    src = fetchFromGitLab {
+      owner = "nmigen";
+      repo = "nmigen";
+      hash = "sha256-tpcA+FFHhm4gF37Z+rB/uZopSRtAtNxU8j5IXnSGeCg=";
+      rev = "29dec30f628356828aa2aa2b91ce205a570d664e"; # HEAD @ version date
+    };
```

Comment:
upstream nmigen has been renamed to https://github.com/amaranth-lang/amaranth. amaranth is packaged in nixpkgs but we can't just override a few of the attributes the way we did for pyelftools because the names are different, so much of this is copied from the amaranth build recipe